pkg/registry/core/pod/strategy.go

```diff
@@ -631,9 +631,14 @@ func streamLocation(
 		return nil, nil, err
 	}
 	params.Add("container", container)
+
+	host := nodeInfo.Hostname
+	if len(nodeInfo.Port) != 0 {
```

Comment:
Under what circumstances will the port be empty?

Reply:
when apiendpoint of cluster is kubernetes.default.svc.cluster.local rather than ip:port, we should not join the host and an empty port.